server/src/main/java/org/elasticsearch/common/joda/Joda.java

```diff
@@ -65,25 +76,25 @@ public static JodaDateFormatter forPattern(String input) {
         }
 
         DateTimeFormatter formatter;
-        if ("basicDate".equals(input) || "basic_date".equals(input)) {
```

Comment:
in progress - this method slightly changes to the one in `DateFormatters` so a different regexp should be used to refactor this